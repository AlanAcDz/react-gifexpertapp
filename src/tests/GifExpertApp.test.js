import React from 'react';
import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp';
import '@testing-library/jest-dom';

describe('<GifExpertApp/> tests', () => {
    test('should render properly', () => {
        const wrapper = shallow(<GifExpertApp/>);
        expect(wrapper).toMatchSnapshot();
    });
    test('should show category list', () => {
        const categories = ['Game of Thrones', 'Bloodborne'];
        const wrapper = shallow(<GifExpertApp defaultCategories={ categories }/>);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid')).toHaveLength(categories.length);
    });
});
