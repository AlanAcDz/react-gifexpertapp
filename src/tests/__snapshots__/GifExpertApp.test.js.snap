// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifExpertApp/> tests should render properly 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`<GifExpertApp/> tests should show category list 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Game of Thrones"
      key="Game of Thrones"
    />
    <GifGrid
      category="Bloodborne"
      key="Bloodborne"
    />
  </ol>
</Fragment>
`;
